vm/interpreter/src/default_runtime.rs

```diff
@@ -265,20 +270,17 @@ where
         };
         self.caller_validated = false;
 
-        // snapshot state tree
-        let snapshot = self
-            .state
-            .snapshot()
-            .map_err(|e| actor_error!(fatal("failed to create snapshot {}", e)))?;
-
         let send_res = vm_send::<BS, SYS, R, P>(self, &msg, None);
 
         // Reset values back to their values before the call
         self.vm_msg = prev_msg;
         self.caller_validated = prev_val;
+        self.state
```

Comment:
I think this is wrong. clear_snapshot should be called after revert_to_snapshot.

Reply:
Just tested the conformance tests with what i said and 3 more tests pass.

Reply:
The way they have their snapshots setup is pretty backwards, their snapshot revert clears a snapshot from their slice then adds a blank one, maybe why the confusion (I don't necessarily think this is a good pattern, it's redundant)